python/test/test_dataset.py

```diff
@@ -482,6 +492,38 @@ def test_Workspace2D_example(self):
                 spec[Data.Value, "sample1"] = np.zeros(1000)
                 spec[Data.Variance, "sample1"] = np.zeros(1000)
 
+    def test_monitors_example(self):
+        d = Dataset()
+
+        d[Coord.SpectrumNumber] = ([Dim.Spectrum], np.arange(1, 101))
+
+        # Add a (common) time-of-flight axis
+        d[Coord.Tof] = ([Dim.Tof], np.arange(9))
+
+        # Add data with uncertainties
+        d[Data.Value, "sample1"] = ([Dim.Spectrum, Dim.Tof], np.random.exponential(size=100*8).reshape([100, 8]))
+        d[Data.Variance, "sample1"] = d[Data.Value, "sample1"]
+
+        # Add event-mode beam-status monitor
+        status = Dataset()
+        status[Data.Tof] = ([Dim.Event], np.random.exponential(size=1000))
+        status[Data.PulseTime] = ([Dim.Event], np.random.exponential(size=1000))
+        d[Coord.Monitor, "beam-status"] = ([], status)
+
+        # Add position-resolved beam-profile monitor
+        profile = Dataset()
+        profile[Coord.X] = ([Dim.X], [-0.02, -0.01, 0.0, 0.01, 0.02])
+        profile[Coord.Y] = ([Dim.Y], [-0.02, -0.01, 0.0, 0.01, 0.02])
+        profile[Data.Value] = ([Dim.Y, Dim.X], (4,4))
+        # Monitors can also be attributes, so they are not required to match in operations
+        d[Attr.Monitor, "beam-profile"] = ([], profile)
+
+        # Add histogram-mode transmission monitor
+        transmission = Dataset()
+        transmission[Coord.Energy] = ([Dim.Energy], np.arange(9))
+        transmission[Data.Value] = ([Dim.Energy], np.random.exponential(size=8))
+        d[Coord.Monitor, "transmission"] = ([], ())
```

Comment:
These seem to be more usage examples than "unit tests" and would eventually need to be re-homed accordingly. Also looks like this is 3 tests in one, maybe different named methods?

Reply:
Would be nice to see these monitors used/accessed as well as defined.

Reply:
I am currently working on an actual larger example where monitors are really used.